numba/tests/test_linalg.py

```diff
@@ -1558,5 +1566,139 @@ def check(a, **kwargs):
                                             dtype=np.float64),))
 
 
+class TestLinalgSlogdet(TestLinalgBase):
+    """
+    Tests for np.linalg.slogdet.
+    """
+
+    @needs_lapack
+    def test_linalg_slogdet(self):
+        """
+        Test np.linalg.slogdet
+        """
+        cfunc = jit(nopython=True)(slogdet_matrix)
+
+        def check(a, **kwargs):
+            expected = slogdet_matrix(a, **kwargs)
+            got = cfunc(a, **kwargs)
+
+            # As numba returns python floats types and numpy returns
+            # numpy float types, some more adjustment and different
+            # types of comparison than those used with array based
+            # results is required.
+
+            # check that the returned tuple is same length
+            self.assertEqual(len(expected), len(got))
+            # and that length is 2
+            self.assertEqual(len(got), 2)
+
+            # check that the domain of the results match
+            for k in range(2):
+                self.assertEqual(
+                    np.iscomplexobj(got[k]),
+                    np.iscomplexobj(expected[k]))
+
+            # turn got[0] into the same dtype as `a`
+            # this is so checking with nulp will work
+            got_conv = a.dtype.type(got[0])
+            np.testing.assert_array_almost_equal_nulp(
+                got_conv, expected[0], nulp=10)
+            # compare log determinant magnitude with a more fuzzy value
+            # as numpy values come from higher precision lapack routines
+            resolution = 5 * np.finfo(a.dtype).resolution
+            np.testing.assert_allclose(
+                got[1], expected[1], rtol=resolution, atol=resolution)
+
+            # Ensure proper resource management
+            with self.assertNoNRTLeak():
+                cfunc(a, **kwargs)
+
+        # test: 1x1 as it is unusual, 4x4 as it is even and 7x7 as is it odd!
```

Comment:
Since you are testing exactly the same stuff, why not mutualize the det() and slogdet() tests?